test/BuyTshirt.spec.ts

```diff
@@ -1,44 +1,62 @@
-import { $, browser } from 'protractor';
+import { browser } from 'protractor';
+import { MenuContentPage } from '../src/page';
+import { AddressStepPage } from '../src/page';
+import { BankPaymentPage } from '../src/page';
+import { OrderResumePage } from '../src/page';
+import { PaymentStepPage } from '../src/page';
+import { ProductAddedModalPage } from '../src/page';
+import { ProductDetailPage } from '../src/page';
+import { ProductListPage } from '../src/page';
+import { ShippingStepPage } from '../src/page';
+import { SignInStepPage } from '../src/page';
+import { SummaryStepPage } from '../src/page';
 
 describe('Buy a t-shirt', () => {
+
+ const menuContentPage: MenuContentPage = new MenuContentPage();
+ const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
+ const addressStepPage: AddressStepPage = new AddressStepPage();
+ const orderResumePage: OrderResumePage = new OrderResumePage();
+ const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+ const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+ const productDetailPage: ProductDetailPage = new ProductDetailPage();
+ const productListPage: ProductListPage = new ProductListPage();
+ const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+ const signInStepPage: SignInStepPage = new SignInStepPage();
+ const summaryStepPage: SummaryStepPage = new SummaryStepPage();
+ 
  beforeEach(() => {
-   jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
+  jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
  });
 
  it('then should be bought a t-shirt', async () => {
    await browser.get('http://automationpractice.com/');
-   await(browser.sleep(10000));
-   await $('#block_top_menu > ul > li:nth-child(3) > a').click();
    await(browser.sleep(3000));
-   await
-   $('#center_column > ul > li > div > div.left-block > div > a.product_img_link > img').click();
+   await menuContentPage.goToTShirtMenu();
```

Comment:
agrega la preparacion de datos a un before, el it sólo se debe encargar de validaciones